clang/lib/Sema/SemaTemplate.cpp

```diff
@@ -3250,8 +3251,7 @@ checkBuiltinTemplateIdType(Sema &SemaRef, BuiltinTemplateDecl *BTD,
 
     TemplateArgument NumArgsArg = Converted[2];
     if (NumArgsArg.isDependent())
-      return Context.getCanonicalTemplateSpecializationType(TemplateName(BTD),
-                                                            Converted);
+      return QualType();
```

Comment:
Can you explain what is happening here?  Why are we no longer returning a type?

Reply:
This just feeds into the CanonType / UnderlyingType of the TemplateSpecializationType created within CheckTemplateId, and returning a null qualtype here has the same effect; it will create a canonical TST with this name and arguments, but is less error prone and there is better sanity checking.

Reply:
Ah, i think i see now.  Thanks!  I'm good now.  Feel free to commit after the above reordering